generic/htx_test.py

```diff
@@ -40,6 +42,29 @@ class HtxTest(Test):
     :params time_limit: how much time(hours) you want to run this stress.
     """
 
+    def install_latest_htx_rpm(self):
+        """
+        Search for the latest htx-version for the intended distro and
+        install the same.
+        """
+        distro_pattern = "%s%s" % (
+            self.dist_name, self.detected_distro.version)
+        temp_string = process.getoutput(
+            "curl --silent %s" % (self.rpm_link),
+            verbose=False, shell=True, ignore_status=True)
+        matching_htx_versions = re.findall(
+            r"(?<=\>)htx\w*[-]\d*[-]\w*[.]\w*[.]\w*", str(temp_string))
+        distro_specific_htx_versions = [
+            htx_rpm for htx_rpm in matching_htx_versions
+            if distro_pattern in htx_rpm]
+        distro_specific_htx_versions.sort(reverse=True)
+        self.latest_htx_rpm = distro_specific_htx_versions[0]
+
+        if process.system('rpm -ivh --nodeps %s%s '
+                          '--force' % (self.rpm_link, self.latest_htx_rpm),
+                          shell=True, ignore_status=True):
+            self.cancel("Installion of rpm failed")
+
```

Comment:
code wise it looks good to me.
@PraveenPenguin @rohan-r-deshpande can we move this part to common code.
we will be needing this in most of our testing like, for block devices, NIC/vNIC
So I feel, it should be like just pass distro version and then it should be able to install the respective HTX RPM.
if not exact to the util part, can we have this in locally in avocado-misc-tests itself?


@abdhaleegit @vaishnavibhat ^^

Reply:
@Naresh-ibm yes  good idea, if @rohan-r-deshpande you not started  we can pursue this as an  enhancement  as this logic we need tp propagate in other area as well

Reply:
@rohan-r-deshpande ah ok then we can pursue this as enhancement, @Naresh-ibm is this looks ok to you?